pkg/datasource.go

```diff
@@ -143,8 +143,13 @@ func (plugin *GrafanaAzureDXDatasource) QueryData(ctx context.Context, req *back
 		return nil, err
 	}
 
+	email := "Noninteractive"
+	if req.PluginContext.User != nil {
+		email = req.PluginContext.User.Email
```

Comment:
@joshhunt @mckn It looked to me like this should work, but inside the docker-compose container, with only the admin user, this always resulted in either `nil` or `"" (I'm extremely rusty with go at this point and can't remember the semantics). Is there a more realistic way to test this?

Reply:
When investigating this I managed to find a bug in Grafana. The PluginContext.User.Email is not set correctly. I've created a [pr](https://github.com/grafana/grafana/pull/30249) for that, but unfortunately that won't be available to the adx plugin until next release of Grafana. Can this PR wait for that? If not, we can use PluginContext.User.Login or PluginContext.User.Name. At least for [Grafana auth](https://grafana.com/docs/grafana/latest/auth/overview/#grafana-auth), PluginContext.User.Login will always be set. 

Thanks for finding the bug!

Reply:
Actually, I think we should always use PluginContext.User.Login since it will always be set. So please update the code to use that instead.

Reply:
Glad I could help with the bug discovery!

Sounds like a plan! I think any distinct value per user would be a major improvement. I'll push up this change momentarily...

Reply:
@sunker how does this look now?